### ISSUE TITLE
issue on catching exceptions inside listener

### ISSUE THREAD

**Issue by gedaiu** [allure-framework/allure1#832]
Hi i just noticed if we raise an exception inside a LifecycleListener    testNG does not stop running the tests. Can we throw these exceptions instead of just log them?

https://github.com/allure-framework/allure-core/blob/8bc2816ecc8f4243668d133e78b3ded444e8b74f/allure-java-adaptor-api/src/main/java/ru/yandex/qatools/allure/experimental/ListenersNotifier.java#L112

**Comment by baev** [comment]
@gedaiu Hi!

Sorry for the delay. `LifecycleListener` should not affect test execution at all. Why do you need such feature?

**Comment by gedaiu** [comment]
because I had to implement a failure detection mechanism, using the listener and there is no method for stopping the tests from the listener. It would be good for my issue to capture `TestFailureException` or something like this to stop the tests. 

Right now I did a dirty hack, by writing a custom listener and using reflection to replace the static final property. It's not bood.. but it works. It would be awesome to be able to achieve this without this hack...
